synapse/handlers/message.py

```diff
@@ -581,6 +583,9 @@ async def create_event(
         # Strip down the auth_event_ids to only what we need to auth the event.
         # For example, we don't need extra m.room.member that don't match event.sender
         if auth_event_ids is not None:
+            # If auth events are provided, prev events must be also.
+            assert prev_event_ids is not None
```

Comment:
I discussed this with @MadLittleMods, although the docstring says that `prev_event_ids` will be computed if not given this happens in the call to `create_new_client_event` ~20 lines below this.

Reply:
As discussed at https://matrix.to/#/!XaqDhxuTIlvldquJaV:matrix.org/$J_SAM4WacKXqTfcSwGfBIcos5vEJO1i6x9L38oyKPY0?via=matrix.org&via=vector.modular.im&via=envs.net